src/ruby/meetings/app/models/means_of_influence.rb

```diff
@@ -2,15 +2,20 @@ class MeansOfInfluence < ApplicationRecord
   def self.types
     %w(Meeting Hospitality Gift Travel)
   end
-  validates :month,:year,presence: true
+  validates :month, :year, :source_file_id, :source_file_line_number, presence: true
   has_many :influence_government_office_people
   has_many :influence_organisation_people
-  has_many :organisations,through: :influence_organisation_people, source: 'organisation'
-  has_many :organisation_people,through: :influence_organisation_people,source: 'person'
-  has_many :government_people,through: :influence_government_office_people,source: 'person'
-  has_many :government_offices,through: :influence_government_office_people, source: 'government_office'
+  has_many :organisations, through: :influence_organisation_people, source: 'organisation'
+  has_many :organisation_people, through: :influence_organisation_people, source: 'person'
+  has_many :government_people, through: :influence_government_office_people, source: 'person'
+  has_many :government_offices, through: :influence_government_office_people, source: 'government_office'
+
+  belongs_to :source_file
 
   self.types.each do |klass|
     scope klass.underscore.downcase.pluralize.to_sym,->{where(type: klass)}
   end
+
+  default_scope -> { order(year: :asc,month: :asc) }
```

Comment:
Yes, there are a whole bunch of stylistic issues.  I fixed a few but I think the real long-term solution is to set up automatic linting via https://houndci.com/ or similar.